src/fulltrade/FullTradeCard.js

```diff
@@ -33,32 +36,34 @@ export default class FullTradeCard extends Component {
 
     render() {
         const { actions, index, marketIsOpen, trade, ticks } = this.props;
-        const { symbolName, lastBoughtContract } = trade;
+        const { lastBoughtContract } = trade.purchaseInfo;
+        const { symbolName } = trade.params;
 
         const contract = this.props.contract || mockedContract;
 
         const contractAllowStartLater = findDeep(contract, child => child && !!child.forwardStartingDuration);
 
         const disabled =
             contract === mockedContract ||
-            trade.disabled ||
+            trade.uiState.disabled ||
```

Comment:
I think you are moving in the wrong direction.
We need to make a better differentiation than a 'trade panel' and a trade object.
Trade object will be flat and has only stuff you need to perform the trade.
Trade panel can be what you are trying to do here.